pages/home.tsx

```diff
@@ -1,11 +1,98 @@
-import { ServerPageProps } from '../types/types'
-import { Box, Text } from 'grommet'
+import { ServerPageProps, Status } from '../types/types'
+import {
+  Box,
+  Text,
+  Tabs,
+  Tab,
+  Card,
+  CardHeader,
+  CardBody,
+  CardFooter,
+  Button,
+  Layer
+} from 'grommet'
+import { useEffect, useState } from 'react'
+import { PrescriptionAndLocation, PharmacyLocation } from '../types/types'
+import QRCodeModal from '../components/QrCode'
+import PatientPrescriptions from '../components/PatientPrescriptions'
+import PharmacyLocationView from '../components/PharmacyLocation'
+import { Prescription } from '.prisma/client'
+import Skeleton from 'react-loading-skeleton'
+
 const Home = ({ user }: ServerPageProps) => {
+  const [prescriptions, setPrescriptions] = useState<
+    Array<PrescriptionAndLocation>
+  >([])
+  const [previousPrescriptions, setPreviousPrescriptions] = useState<
+    Array<PrescriptionAndLocation>
+  >([])
+  const [pharmacyLocation, setPharmacyLocation] = useState<PharmacyLocation>()
+  const [location, setLocation] = useState<Location>()
+  const [loading, setLoading] = useState(true)
+  const [qrCode, setQrCode] = useState('')
+  const [showQrCode, setShowQrCode] = useState(false)
+
+  const fetcher = (userRole: string) => {
+    if (userRole === 'patient') {
+      return fetch('/api/prescriptions/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          patientId: user.Patient?.id
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          let prevPrescriptions: PrescriptionAndLocation[] = []
+          let activePrescriptions: PrescriptionAndLocation[] = []
+
+          data.prescription?.forEach(
+            (prescription: PrescriptionAndLocation) => {
+              if (prescription.status === Status.AwaitingPickup) {
+                activePrescriptions.push(prescription)
+              } else if (prescription.status === Status.PickupCompleted) {
+                prevPrescriptions.push(prescription)
+              }
+            }
+          )
+          setPreviousPrescriptions(prevPrescriptions)
+          setPrescriptions(activePrescriptions)
+        })
+        .catch()
+        .finally(() => setLoading(false))
+    } else if (userRole === 'staff') {
+      return fetch('/api/locations/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          id: user.Staff?.locationId
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          setPharmacyLocation(data.location)
+        })
+        .catch()
+        .finally(() => setLoading(false))
+    }
+  }
+
+  const { data, error } = useSWR(user.role, fetcher)
+
+  const onClose = () => setShowQrCode(false)
+
+  if (loading && !error) {
+    return (
+      <Box gap="medium">
+        <Skeleton count={1} height={30} />
+        <Skeleton count={5} height={30} style={{ lineHeight: '3' }} />
+      </Box>
+    )
+  }
```

Comment:
because you are doing this at the page level. we are blocking access to the rest of the application. If you move the fetching into the components, then you can resolve this "full page load" issue
<img width="1440" alt="Screen Shot 2022-11-06 at 1 18 24 AM" src="https://user-images.githubusercontent.com/33642354/200155495-881e8e1c-f743-4e71-b474-dfc5a5bd5f77.png">